File: backend/src/TripPlanning.Service/Models/Expense.cs
namespace TripPlanning.Service.Models;

public enum ExpenseCategory
{
    Transport,
    Accommodation,
    Food,
    Tickets,
    Shopping,
    Other,
}

public class Expense
{
    public Guid Id { get; set; }
    public Guid TripId { get; set; }
    public required string Name { get; set; }
    public ExpenseCategory Category { get; set; } = ExpenseCategory.Other;
    public decimal Amount { get; set; }
    public DateOnly Date { get; set; }
    public string Description { get; set; } = string.Empty;

    public Trip? Trip { get; set; }
}

File: frontend/src/models/Expense.ts
export type ExpenseCategory = "Transport" | "Accommodation" | "Food" | "Tickets" | "Shopping" | "Other";

export interface Expense {
  id: string;
  tripId: string;
  name: string;
  category: ExpenseCategory;
  amount: number;
  date: string;
  description: string;
}
